pkg/api/message/v1/service.go

```diff
@@ -79,6 +80,38 @@ func NewService(node *wakunode.WakuNode, logger *zap.Logger) (s *Service, err er
 	return s, nil
 }
 
+func (s *Service) pollForMessages() {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	store, ok := s.waku.Store().(*store.XmtpStore)
+	if !ok {
+		panic("could not start store")
+	}
+
+	lastChecked := time.Now()
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-ticker.C:
+			newLastChecked := time.Now()
+			s.log.Info("checking for messages", zap.Time("since", lastChecked))
+			res, err := store.FindMessagesSince(lastChecked.UnixNano())
+			if err != nil {
+				s.log.Error("error querying store", zap.Error(err))
+				continue
+			}
+			if s.dispatcher != nil {
+				for _, msg := range res {
+					s.dispatcher.Submit(msg.ContentTopic, buildEnvelope(msg))
+				}
+			}
+			lastChecked = newLastChecked
```

Comment:
Wondering if there is a race condition here if a publish and query happen at the same time that might cause occasional messages to be skipped. Would it make sense to use the receiver timestamp of the latest message fetched here?

Reply:
Ya. I definitely share your concerns about races. Even with the last message's timestamp, you could still get missed messages (clock drift between nodes).

I was thinking about adding some sort of padding to the times and just keeping track of recent message IDs to dedupe.